In progress - CI-related - FREQUENT TRIGGERS - IGNORE - DO NOT MERGE

heinsaar opened the pull request:
#### Testing

Testing - ignore

dmaranges-gd committed:
Add codecov code coverage reports to PRs (#38342)

* Add new yaml file to workflow actions to upload coverage of unit tests to codecov

* Add new steps to new yaml

* Delete unnecessary jobs

* Add new job to try to send data coverage to codecov

* Delete extra lines

* Add upload command

* Delete extra lines and add a command

* Edit rm command to clean environment

* Edit clean commands

* Add coverage to other tests to load on codecov

* Add new files to coverage data about different tests

* Skip python environment setup

* Delete coverage file

* Delete coverage workflow file and add job to build.yaml

* Restore some lines affected by automatic formatting

* Fix a typo and delete duplicated lines

andy31415 committed:
Disable flaky Telink CI (#38884)

wy-hh committed:
[bouffalo lab] Update reference partition table to increase size of P… (#38875)

* [bouffalo lab] Update reference partition table to increase size of PSM region

* Restyled by prettier-markdown

---------

Co-authored-by: Restyled.io <commits@restyled.io>

andy31415 committed:
Revert "Add codecov code coverage reports to PRs (#38342)" (#38889)

This reverts commit 6876fcb8e1f1fd7489e6189a4179cb3bab9db493.

cecille committed:
TC-TMP-2.1: Allow tolerance to be optional (#38771)

Harshith-GRL committed:
Return python object intead of bool value as Response for commissioning with device (#38787)

* Return python object intead of bool value as Response for commissioning with device

* added docstrings PairingStatus class

* Restyled by autopep8

* set the PairingStatus to False when exception occurs

* Updated PairingStatus class to evaluate to true without success parameter to make it easier to use

* bool expression validation is made directly instead of storing in variable and if success do not return empty string

* added new line to re-start CI job

---------

Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: cjandhyala <68604034+cjandhyala@users.noreply.github.com>

andy31415 committed:
Make sure we follow up with any deletions in zap templates check (#38802)

* Add ci test for clean generation of clusters

* Remove DRLC, fix up zap templates check

* Remove extra file

* Add back file from master

* Delete DRLC

* File not generated anymore, should be removed

* Regen

dinabenamar committed:
[NXP] Adding CMake build system support for NXP FreeRTOS platforms (#38518)

* [NXP][cmake] Initial enablement of Matter with SDK 3.0 to build NXP examples with cmake

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][cmake] Updating cmake to support new matter stack from 04/14/2025

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

* [NXP][cmake] Updating application CMakeLists.txt to display error when building with wifi + thread simultaneously

* Supporting WiFi + Thread simultaneously requires enabling the TBR cluste in the application ZAP file.
* Therefore, we must ensure that this configuration is not built on applications that do not support it.

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][build_examples] Update the NXP builder scripts to support building with the new cmake build system

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][workflows] Enable Matter CMake examples in NXP build workflows

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][docs][common] Rework documentation for NXP examples

* Create a dedicated readme common to freertos platforms under "docs/platforms/nxp/"
* Move all-clusters & thermostat doc under common folder to keep only application relevant information
* Add information about freertos cmake config files in "config/nxp/README.md"

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

[docs] Adding documentation to describe the application build system architecture of NXP freertos applications

* Application build system architecture
* Provide customization options

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

[docs][common] Update index.md with new NXP documentation

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

[nxp toup] [doc] Adding link to OTA Software update guides

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

* [NXP][doc][rw61x][rt1170][rt1060] Rework documentation for NXP RW/RT examples

* Move RW/RT platform-specific information into a common guide under "docs/platforms/nxp"
* Remove RW/RT all-clusters and thermostat readme to avoid duplication

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][doc][RTS] Updating ota software update readme to reference new sdk_next path and multi-image OTA support

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

* [NXP][docs][zephyr] Rework documentation for NXP Zephyr platform

* [NXP][examples][docs][rw61x] Update laundry-washer readme with reference to new platform guide

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][examples][docs][rt1170] Update laundry-washer readme with reference to new platform guide

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][examples][docs][rt1060] Update laundry-washer readme with reference to new platform guide

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][docs][common] Update factory data info for CMake support

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][doc][common] Update in all-clusters-app readme

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][cmake] Updating copyrights year date

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP] Update requirements.nxp.txt to make sure CMake version is >=3.30 when build NXP examples

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP] Revert unwanted changes in config/nxp/chip-gn-freertos/args.gni with dos2unix

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* Restyled by whitespace

* Restyled by gn

* Restyled by prettier-markdown

* Restyled by prettier-yaml

* Restyled by autopep8

* [NXP] Fix spelling errors in doc

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* [NXP][workflows] Removing artifact copy for freeRTOS cmake targets

+ scripts/build/testdata/all_targets_linux_x64.txt alignement to add new NXP variants

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

* [NXP][build_examples] Switching to FRDM board by default for RW platform

+ scripts/build/testdata/all_targets_linux_x64.txt alignement to add new NXP variants

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

* Update NXP build example

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Restyled by autopep8

* Update NXP build example

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Update nxp.py

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Restyled by autopep8

* Update build examples target argument for cmake build

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Update nxp_matter_support submodule

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

---------

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>
Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>
Signed-off-by: Martin Girardot <martin.girardot@nxp.com>
Co-authored-by: Gatien Chapon <gatien.chapon@nxp.com>
Co-authored-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>
Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: Martin Girardot <martin.girardot@nxp.com>

andy31415 committed:
Add sufficient information in matter idl to reconstruct acronym/non-acronym logic on constant names (#38888)

* Add the ability to generate specification name from matter idl

* Support parsing of specification names

* Updated example a bit

* zap regen

* Update with real world examples

pidarped committed:
Add support for 'DynamicConstraintError' status code. (#38807)

sabollim-silabs committed:
[Closure] Refactoring closure control server implementation  (#38545)

* Closure control changes

* Restyled by whitespace

* Restyled by clang-format

* Add Closure Control unittest cases

* Restyled by whitespace

* Restyled by clang-format

* Resolve build error in CI

* Resolve CI build errors

* Add missing source files

* Addressing PR comments

* Adding UnitTests

* Restyled by whitespace

* Restyled by clang-format

* Addressing PR comments

* Fix build CI failures

* Restyled by whitespace

* Restyled by clang-format

* Restyled by gn

* refactoring setter for OverallState and OverallTarget

* Restyled by whitespace

* Restyled by clang-format

* Fix CI failures

* Fix CI failures

* Restyled by clang-format

* addressing PR comments

* Restyled by whitespace

* Restyled by clang-format

* Apply suggestions from code review

Co-authored-by: Boris Zbarsky <bzbarsky@apple.com>

* Addressing PR comments

* Restyled by whitespace

* Restyled by clang-format

* rearranging private methods and variables

* Restyled by clang-format

* Removing delegate specific reporting in countdowntime

* Update function comments

* Restyled by whitespace

* Restyled by clang-format

* Removing Testcases related to COundowntime changes

* Fix closure app failure with closure control server changes

* Restyled by clang-format

* Removing commented code

* Added TODO for countdowntime handling

* Addressing PR comments

* Restyled by whitespace

* Restyled by clang-format

---------

Co-authored-by: sabollim <satyanaag.bollimpalli@silabs.com>
Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: Arun Padakanti <arunkumar.padakanti@silabs.com>
Co-authored-by: Boris Zbarsky <bzbarsky@apple.com>

vijs committed:
[E2E][JF] Joint Fabric Datastore Cluster updates (#38880)

* Addressed Joint Fabric Datastore Cluster spec updates

* Added files generated from JF Datastore Cluster updates

vijs committed:
[E2E][JF] Joint Fabric Datastore Cluster updates (#38880)

* Addressed Joint Fabric Datastore Cluster spec updates

* Added files generated from JF Datastore Cluster updates

heinsaar committed:
`CI` Try to fail

heinsaar committed:
`CI` Trigger again (test)

tcarmelveilleux committed:
Remove irrelevant and poorly configured localization clusters from some samples (#38885)

* Remove localization config cluster from some examples

* Remove time format localization cluster from some examples

* Remove unit localization cluster from some examples

* ZAP regen

* Re-enable some tests

* Re-disable LTIME-3.1 until fixed

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=38903) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=38903) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=38903) it.</sub>

heinsaar committed:
Restore (test)

pgregorr-amazon committed:
tv-casting-app/tv-app target app, commissioner passcode flow (#38856)

tcarmelveilleux committed:
Bump some cluster revisions in examples (#38905)

* Update General Commissioning Version in XML

* ZAP regen

* Restyled by clang-format

* Remove invalid constraints from matter-devices.xml

* Bump RootNode/Aggregator/Dishwasher dev type revs and regen

* Made ClusterRevision External on General Commissioning

* Fix TestDescriptorCluster

---------

Co-authored-by: Restyled.io <commits@restyled.io>

bzbarsky-apple committed:
Stop running the MTRDevice dealloc test when pool width is not 1. (#38854)

This fails intermittently in testSubscriptionPoolManyDevices, but more
importantly: this dealloc test only makes sense in the width-1 case, because if
the pool width is not 1 then the test is not testing anything (because nothing
gets blocked on the one badly behaving thing if it does behave badly).

bzbarsky-apple committed:
Add BLE APIs to allow commissioning using concatenated QR codes. (#38922)

For now not implemented; platforms will need to provide implementations.

bzbarsky-apple committed:
Fix typo in log message. (#38926)

ksperling-apple committed:
Guard defines in standalone/CHIPProjectConfig.h (#38928)

* Guard defines in standalone/CHIPProjectConfig.h

... so individual apps' CHIPProjectAppConfig.h can override these settings if
needed without triggering macro redefinition warnings / errors.

* Get rid of unused define CHIP_CONFIG_ENABLE_UPDATE

* Don't guard CHIP_CONFIG_SECURITY_TEST_MODE

ksperling-apple committed:
Use SOURCE_DATE_EPOCH from environment if present (#38929)

Update the script that generates FirmwareBuildTime.h to use the timestamp from
the SOURCE_DATE_EPOCH environment if it is present and ahead of the time stored
in the fallback_last_known_good_time file. This allows the timestamp to be
easily picked up from build systems that follow this standard for supplying the
source timestamp (https://reproducible-builds.org/specs/source-date-epoch/)

Also do the conversion between POSIX and Matter epoch times by simple
offsetting, instead of dealing with time zones and calendars.

dinabenamar committed:
[NXP][cmake][common] Provide a way to customize the toolchain in the NXP freertos cmake abstraction (#38904)

* [NXP][cmake][common] Provide a way to customize the toolchain in the freertos cmake abstraction

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>

* Restyled by gn

---------

Signed-off-by: Dina Benamar <dina.benamarelmaaroufi@nxp.com>
Co-authored-by: Restyled.io <commits@restyled.io>

wy-hh committed:
[Bouffalo lab] ‌Improve BL702/BL702L execution efficiency by updating the linker file‌ (#38878)

* [Bouffalo lab] configure memory region of Thread task stack on BL702/BL702L

* Restyled by clang-format

---------

Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: Andrei Litvin <andy314@gmail.com>

gabc-cpp committed:
[codegen/CodegenDataModelProvider.cpp]: Move the constructor of AttributeEntry inside a closer "for" loop. (#38866)

* Moved AttributeEntry constructor inside of "for" loop.

* Using move semantics to handle object "globalListEntry". Renaming "attribute".

kiel-apple committed:
Disable flaky Darwin tests (#38795)

* disable `test034_TestMTRDeviceHistoricalEvents`

* disable all `MTRPerControllerStorageTests`

`testSubscriptionPoolManyDevices` is more flaky than others but exercising additional caution

* add exemplars for `TestMTRDeviceHistoricalEvents` flake

* add notes to `testMTRDeviceDealloc` disable

* Revert "disable all `MTRPerControllerStorageTests`"

This reverts commit dfceea9a24e2e1f66ad973b7cf56433dd2f7b943.

* re-enable `test034_TestMTRDeviceHistoricalEvents` - fix landed

* try disabling subscription pool tests as well

* add issue links for subscription pool flakes

wy-hh committed:
[bouffalo lab] fix bl616 wifi scan operation with/without specified ssid (#38877)

* [bouffalo lab] fix bl616 wifi scan operation with/without specified ssid

* Restyled by clang-format

* check ssid length before

* Verify SSID length prior to copying and correct pScanResult reference handling when SSID is specified.

* Restyled by clang-format

* remove +1 for mScanSSID

* Restyled by clang-format

---------

Co-authored-by: Restyled.io <commits@restyled.io>

arkq committed:
Install matter-yamltests files in matter namespace (#38534)

* Install matter-yamltests files in matter namespace

* Update imports in tests

* Update yamltests imports

* pytests: Include py_matter_idl in venv

Include matter.idl when pytest deps are requested.
Add pytest deps to venv used to run chip-repl tests.

Signed-off-by: Maciej Grela <m.grela@samsung.com>

* Update tests requirements

* Install testing requirements in python-venv

* Add tests requirements to cache key

* Install matter-yamltests in cert-bins docker image

* Update matter-yamltests sources

---------

Signed-off-by: Maciej Grela <m.grela@samsung.com>
Co-authored-by: Maciej Grela <m.grela@samsung.com>

vivien-apple committed:
[Darwin][Network.framework] Do not include TCPEndPointImpl.h if TCP endpoint are not enabled (#38909)

vivien-apple committed:
[Darwin] Install dispatch source handler for SIGINT/SIGTERM instead of posix signals to make it more reliable (#38910)

vivien-apple committed:
[Darwin][Network.framework] Split the DnssdHostNameRegistrar such that the interface monitoring bits is accessible for other usage (#38911)

vivien-apple committed:
[Darwin][Network.framework] Ensure CHIP_SYSTEM_CONFIG_MULTICAST_HOMING is enabled if chip_system_config_use_network_framework is enabled (#38913)

peterv-apple committed:
Add new test to check that _deviceMayBeReachable switches the device's state to notReachable. (#38772)

Martin-NXP committed:
[NXP][docker] Update NXP Zephyr docker image (#38954)

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

Martin-NXP committed:
[NXP][docker] Update NXP Zephyr docker image (#38954)

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

gd-mauri committed:
Unit Tests for controller/ExampleOperationalCredentialsIssuer. (#38525)

* proposed in-memory persistent delegate implementation.

* increased coverage and TestPersistentStorageDelegate reused.

* bringing constants back into source file.

* addressing PR comments.

---------

Co-authored-by: Andrei Litvin <andy314@gmail.com>

andy31415 committed:
Decouple software-fault-listener (#38914)

* Start decoupling, just added some basic files

* Update code for decoupling (serach and replace for now ... need to fix dependencies)

* Update deps to public_deps for easier includes

* Fix dependencies

* Fix duplicate namespace

* Restyle

* update for cmake dependencies: needs to pull in the software fault listener

* Fix typos

* Remove SoftwareDiagnosticsTriggerHandler outside the cluster ... it is a source set in platform (a bit of a WTF there really...)

* Remove odd whitespace issue

* Restyled by gn

* Fix include for software diagnostics stub...stop relying on transitive includes

* Another infineon fix

* Fix namespaces

---------

Co-authored-by: Andrei Litvin <andreilitvin@google.com>
Co-authored-by: Restyled.io <commits@restyled.io>

andy31415 committed:
Update camera app to be buildable with clang, add camera app to local.py (#38770)

* Add camera app to local.py

* Fix clang complaing about size_t/ssize_t usage

* Some compile fixes for cameras

* Build updates to support clang

* Undo static casts on members, change member types

* Update based on code review

* Restyle

* use variables instead of constants - review feedback

* Fix int/uint

* cmake4 system fails to build libdatachannel. Fix that too in local.py for now

* Fix casts

* Restyled by autopep8

---------

Co-authored-by: Andrei Litvin <andreilitvin@google.com>
Co-authored-by: Restyled.io <commits@restyled.io>

vijs committed:
[E2E][JF] Made JF Datastore cluster attributes/commands provisional (#38946)

* Made JF Datastore cluster attributes/commands provisional

* Made JF Admin cluster attributes/commands provisional

vijs committed:
[E2E][JF] Made JF Datastore cluster attributes/commands provisional (#38946)

* Made JF Datastore cluster attributes/commands provisional

* Made JF Admin cluster attributes/commands provisional

ksperling-apple committed:
Assign a unique test PID to the network-manager example app (#38930)

bzbarsky-apple committed:
Log the AKID when a PAA is not found during device attestation. (#38964)

This helps determine which PAAs might be missing from a commissioner.

erwinpan1 committed:
[Chef] chef.py supports --build_bundle (#38945)

* chef.py supports --build_bundle

* Restyle

abiradarti committed:
[TI] Dynamic Short/Long Idle Time Support for CC13x4_26x4 (#38921)

* Dynamic Short/Long Idle Time Support for TI CC13x4_26x4

* Restyled by whitespace

* Restyled by clang-format

* Restyled by gn

* Restyled by prettier-markdown

---------

Co-authored-by: Restyled.io <commits@restyled.io>

bzbarsky-apple committed:
Re-enable Darwin tests that have been fixed. (#38972)

This reverts commit 47b3a6aecc28e708ec67adc4d31cfc0412ac65f1.

The subscription pool tests were fixed in #38854.

testMTRDeviceDealloc was fixed in #38809.

Both happened before the PR to disable them got through review and landed.

andy31415 committed:
Disable flaky darwin test - test036_TestStorageBehaviorConfiguration (#38948)

* Darwin test disable

* Update src/darwin/Framework/CHIPTests/MTRDeviceTests.m

brdandu committed:
Cleaning up the package order path such that templates json paths come after zcl json paths for all .zap files when opened and saved (#38958)

Github: ZAP#1126

vijs committed:
Added implementations for the JF Administrator and Datastore Servers (#38842)

cecille committed:
TC-LTIME-3.1: Remove YAML (#38761)

Test was automated in this PR:
https://github.com/project-chip/connectedhomeip/pull/38107/files

YAML is no longer required.

heinsaar committed:
Refactor CI test argument handling

Centralize parsing via `env_test.yaml` and remove redundant discriminator flag. This is meant to build upon and continue https://github.com/project-chip/connectedhomeip/pull/37494

heinsaar committed:
Merge branch 'master' into 38665-remove-redundancy-from-CI

fessehaeve committed:
added a setter for TemperatureUnit attribute to be used by applications (#38976)

* added a setter for TemperatureUnit attribute to be used by applications

* added check for return values

bzbarsky-apple committed:
Revert "[Darwin] Install dispatch source handler for SIGINT/SIGTERM instead of posix signals to make it more reliable (#38910)" (#38992)

This reverts commit 93d1ae26710c04a1533417efc06e8aea33e41f22.

After these changes, `[NSTask terminate]` (which sends SIGTERM) does not
reliably terminate tasks, leading to random test failures.

Fixes https://github.com/project-chip/connectedhomeip/issues/38989

Martin-NXP committed:
[NXP] Add freertos rw61x ethernet support (#38975)

* [NXP][common] Add ethernet support for rw61x platform

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* [NXP] Update documentation for ethernet support

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Regenerate ethernet zap files

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Enable freertos ethernet cmake build

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Restyled by whitespace

* Restyled by clang-format

* Restyled by gn

* Move chip_enable_ethernet args declaration to src/platform/device.gni as it is done for wifi or thread

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* [NXP] Update ethernet zap file

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Restyled by prettier-markdown

* Update doc file

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>

* Restyled by prettier-markdown

---------

Signed-off-by: Martin Girardot <martin.girardot@nxp.com>
Co-authored-by: Restyled.io <commits@restyled.io>

jwhui committed:
[docs] add DeepWiki badge to enable auto-refresh (#38995)

jamesharrow committed:
Misc fixes for TC_EGC and TC_SEPR test cases (#38893)

* Fixes #38892 - should have been lowercase .currentPrice

* Improved TCP testing based on PICS "MCORE.SC.TCP" so that if the DUT doesn't support TCP we can still check basic read attribute with a forecast.

* Fix missing 'TC' in 'desc_TC_EGC_2_1' etc (see https://github.com/project-chip/certification-tool/issues/591)

* Added missing 'TC' into all EGC scripts

* Improved exception handling when failing to establish TCP we get ChipStackError exception (not TimeoutError)

---------

Co-authored-by: Hasty Granbery <granbery@google.com>

gmarcosb committed:
Add support for isFabricScoped on a command & use it in TLSCertificateManagementCluster (#37969)

* Add support for isFabricScoped on command

* Re-introduce isFabricScoped on structs & commands

* Generated using ./scripts/tools/zap_regen_all.py

* Apply code review suggestions

* Improve calls to Decode

* Revert unnecessary change from client-side changes

* Merge conflicts from https://github.com/project-chip/connectedhomeip/pull/38390

* Additional changes to support EncodeForRead with accessingFabricIndex

* Add kIsFabricScoped to hand-rolled structs

* Add VID Verification Statement to TC-RR-1.1 (#38712)

* Add VID Verification Statement to TC-RR-1.1

- Add VID Verification Statement filling to TC-RR-1.1
  - Matches https://github.com/CHIP-Specifications/chip-test-plans/pull/5131
- Add missing logging
- Add missing checks as well that were listed in test plan but not actually
  implemented.
- Add native methods missing to implement test changes

* Restyled by clang-format

* Restyled by autopep8

* Fix lint

* Update src/python_testing/TC_RR_1_1.py

Co-authored-by: C Freeman <cecille@google.com>

* Fix value access in attestation challenge getter

---------

Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: C Freeman <cecille@google.com>

* Add VID Verification Statement to TC-RR-1.1 (#38712)

* Add VID Verification Statement to TC-RR-1.1

- Add VID Verification Statement filling to TC-RR-1.1
  - Matches https://github.com/CHIP-Specifications/chip-test-plans/pull/5131
- Add missing logging
- Add missing checks as well that were listed in test plan but not actually
  implemented.
- Add native methods missing to implement test changes

* Restyled by clang-format

* Restyled by autopep8

* Fix lint

* Update src/python_testing/TC_RR_1_1.py

Co-authored-by: C Freeman <cecille@google.com>

* Fix value access in attestation challenge getter

---------

Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: C Freeman <cecille@google.com>

* Generated using ./scripts/tools/zap_regen_all.py

---------

Co-authored-by: Tennessee Carmel-Veilleux <tennessee.carmelveilleux@gmail.com>
Co-authored-by: Restyled.io <commits@restyled.io>
Co-authored-by: C Freeman <cecille@google.com>

kiel-apple committed:
[darwin] expose diagnostic log download state (#39002)

bzbarsky-apple committed:
Fix random failure in test029_MTRDeviceWriteCoalescing. (#38999)

The idea is to avoid racing the thread enqueing work items and the thread
handling them, so that we don't get "premature" work item dispatch.

Fixes https://github.com/project-chip/connectedhomeip/issues/38997

jtung-apple committed:
[Darwin] MTRDevice needs to clear subscription pool work in edge cases (#39003)

pimpalemahesh committed:
feat: Pass discovery timeout to Matter server initialization (#38940)

swan-amazon committed:
water-leak-detector: add command handling support for leak state testing (#38966)

The water leak detector example app lacked a mechanism for simulating leak state
transitions via external commands, making it difficult to test Matter command
handling or verify attribute changes during automation.

This change introduces a new command processing framework that enables testing
the `BooleanState::StateValue` attribute using named pipe input. Key additions:

- Added `WaterLeakSensorManager` to encapsulate attribute update logic.
- Introduced `WaterLeakAttrUpdateHandler` and delegate to handle incoming JSON commands.
- Modified `main.cpp` to initialize and run the command processor.
- Registered the example in the certification container image and updated its
  build configuration to include required dependencies (e.g., jsoncpp).
- Created a usage README explaining commissioning and state toggling via FIFO.

This enables the attribute value to be toggled externally using simple JSON-based
commands like:

```bash
echo '{"Name":"LeakStatus","NewValue":1}' > /tmp/chip_water_leak_sensor_fifo_*

s07641069 committed:
[Telink] Update Docker image (Zephyr update) (#39022)

* [Telink] Move w91 binaries into Docker

* test

* [Telink] Update Docker image (Zephyr update)

* [Telink] Use telink_w91_post_build for Zephyr v3.3

heinsaar committed:
Fix CI error

Alami-Amine committed:
[WriteClient] Adding UnitTests for PutPreencodedAttribute and other  related methods (#38752)

* [WriteClient] Adding UnitTests for PutPreencodedAttribute and other related methods

* Integrating Comments

* Adding one more testcase in PreEncoded TLV

* deacativating TestWriteChunking for fake platform

* optimising: Putting repeeated testcases in a loop

* Integrating comments

* matching signs in assert

* adding maybe_unused to variable only used in progress logs

tersal committed:
Generate FlowMeasurement cluster with Alchemmy (#38969)

andy31415 committed:
Bug fix: do not try to add global attributes in attribute path expand iterator (#38993)

* Bug fix: do not try to add global attributes in attribute  path expand iterator - metadata is a concern of the provider

* Remove obsolete comment based on review

---------

Co-authored-by: Andrei Litvin <andreilitvin@google.com>

vivien-apple committed:
[Darwin] Add NSManagedObjectModel to the leak suppression (#38977)

j0tunn committed:
style: remove trailing white space (#39023)

yufengwangca committed:
[camera] Use common configs to create a GStreamer pipeline (#39009)

* Use common configs to create a GStreamer pipeline

* Don't mix aspect ratios between the min and max resolutions.

marktrayer committed:
Handle change of DPTZRelativeMove -> DPTZStreams (list of int to list of struct) (#38850)

* XML update of CameraAVStreamManagement from Alchemy

* Zap regen

* Align with updated XML

* Adding Push AV

* Zap regen after adding PushAV

* Restyled by whitespace

* Restyled by clang-format

* Restyled by prettier-json

* Handle watermark in snapshot stream modify
Correctly handle setting of nightvision based on the HAL

* Restyled by whitespace

* Restyled by clang-format

* Revert optattr setting

* Restyled by clang-format

* Align with latest PR that makes enum values contiguous

* Add regen code

* Align with latest camera spec.
Change DPTZRelativeMove attribute to DPTZStreams, which is now a struct that takes a viewport and a stream ID.
Handle plumbing in the app to correctly set values based on stream allocation and deallocation.

* Zap Regen after merge with master

* Restyled by whitespace

* Restyled by clang-format

* Remove debug log

* Zap formatting fixes

* Add static cast

* Remove inadvertant inclusions

* Update AVSUM 2_1 to reflect the new attribute name and type

* Zap changes

---------

Co-authored-by: Restyled.io <commits@restyled.io>

vivien-apple committed:
[Darwin][CI] Fallback to SIGKILL after 10 seconds if terminate fails for NSTask cleanup (#39020)

vivien-apple committed:
[Darwin][Network.framework] Update the interface monitor such that it returns a nw_interface_t instead of the interface index (#38983)

andy31415 committed:
clean obsolete rules: we switched to list builders since then and several notes were invalid (#39032)

Co-authored-by: Andrei Litvin <andreilitvin@google.com>

vivien-apple committed:
[Darwin] Remove src/darwin/Framework/CHIP/BUILD.gn (#39019)

kiel-apple committed:
[Darwin] MTRDevice diagnostic log transfer state improvements (#39033)

* switch to std::lock_guard for locking

* implement `diagnosticLogTransferInProgress` accessor in `MTRDevice_XPC`

* internal property change must notify delegate

* populate diagnostic log transfer state in `_internalProperties`

woody-apple committed:
Fix wrong param (#39040)

jtung-apple committed:
[Darwin] Unit test for #39003 (#39041)

* [Darwin] Unit test for #39003

* Update src/darwin/Framework/CHIPTests/MTRDeviceTests.m

Co-authored-by: Kiel Oleson <kielo@apple.com>

---------

Co-authored-by: Justin Wood <woody@apple.com>
Co-authored-by: Kiel Oleson <kielo@apple.com>

shubhamdp committed:
docs: add environment troubleshooting to esp32 app building guide (#39011)

yufengwangca committed:
Fix CI broken on TC_AVSUM_2_7.py (#39045)

cecille committed:
TC-LTIME-3.1: Fix to not hard code calendars (#38824)

* TC-LTIME-3.1: Fix to not hard code calendars

This was previously done from a PIXIT. Now it reads from the device
and checks all.

* make the get_endpoint call easier to read

* Add test back into CI

heinsaar committed:
Merge branch 'master' into 38665-remove-redundancy-from-CI

heinsaar committed:
Restyle

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=38903) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=38903) before we can accept your contribution.<br/>**30** out of **39** committers have signed the CLA.<br/><br/>:white_check_mark: vijs<br/>:white_check_mark: kiel-apple<br/>:white_check_mark: ksperling-apple<br/>:white_check_mark: gd-mauri<br/>:white_check_mark: tersal<br/>:white_check_mark: tcarmelveilleux<br/>:white_check_mark: bzbarsky-apple<br/>:white_check_mark: pgregorr-amazon<br/>:white_check_mark: shubhamdp<br/>:white_check_mark: jamesharrow<br/>:white_check_mark: andy31415<br/>:white_check_mark: pidarped<br/>:white_check_mark: peterv-apple<br/>:white_check_mark: Martin-NXP<br/>:white_check_mark: wy-hh<br/>:white_check_mark: jtung-apple<br/>:white_check_mark: arkq<br/>:white_check_mark: cecille<br/>:white_check_mark: swan-amazon<br/>:white_check_mark: Harshith-GRL<br/>:white_check_mark: dinabenamar<br/>:white_check_mark: pimpalemahesh<br/>:white_check_mark: woody-apple<br/>:white_check_mark: s07641069<br/>:white_check_mark: sabollim-silabs<br/>:white_check_mark: yufengwangca<br/>:white_check_mark: Alami-Amine<br/>:white_check_mark: fessehaeve<br/>:white_check_mark: gmarcosb<br/>:white_check_mark: marktrayer<br/>:x: j0tunn<br/>:x: heinsaar<br/>:x: brdandu<br/>:x: dmaranges-gd<br/>:x: jwhui<br/>:x: gabc-cpp<br/>:x: abiradarti<br/>:x: erwinpan1<br/>:x: vivien-apple<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=38903) it.</sub>

andy31415 commented:
Closing testing PR that seems stale (no update for almost 1 year, says testing only).
